src/components/GalleryCard.tsx

```diff
@@ -23,27 +20,28 @@ function GalleryCard({
   image: string;
   altText: string;
 }) {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  const createImageURL = (iiifValue: number, imageId: string) => {
-    const url = `https://www.artic.edu/iiif/${iiifValue}/${imageId}/full/843,/0/default.jpg`;
-    return url;
-  };
-
   return (
-    <li>
-      <div className="card">
-        <img alt={altText} src={image} className="art-image"></img>
-        <div className="card-info">
-          <div className="card-all-text">
-            <div className="title-and-artist">
-              <h3 className="card-title">{title}</h3>
-              <p className="artist-name">{artist}</p>
-            </div>
-            <p className="card-location">{location}</p>
+    <div className="card">
+      <img
+        alt={altText}
+        src={image}
+        className="art-image"
+        onError={(e) => {
+          const target = e.target as HTMLImageElement;
+          target.src =
+            "https://eagle-sensors.com/wp-content/uploads/unavailable-image.jpg";
+        }}
```

Comment:
Very cool that you added an `onError` handler. However, a few things:

* You added `src/assets/unavailable-image.jpg` to the project. However, you're not using it here. Instead, you're using a 3rd party URL. Why not use the one from the project?
* It's a really smart idea to use the `target` HTML element object and just set the `src` to the error image. However, …
  * By doing this you're creating a deviation between the React representation (it still has the `src` from the props) and the HTML (it's gonna have the error image src). If the component gets updated React would notice that the HTML is out of sync with the React representation. This could cause issues. In that case it's better to have a _React-way_ of redefining the `src` (i.e. by using a state).
  * What if loading the `unavailable-image.jpg` also fails? In that case the `onError` handler would probably be called again and the `src` would be set to the `unavailable-image.jpg` again which would fail again which would trigger the `onError` handler again which would set the `src` again and so on and so forth … (I didn't try it out. It could be that the browser might not do anything if the `src` is set to a URL that is already present) So, you might create a never ending loop here.

Reply:
OK, thanks for the feedback - makes sense about creating difference between the value React holds & the HTML, so I have rewritten the code using a state.

What you said about the infinite loop was true and happening when accessing the external URL. Now I switched to the image in the project which reduces the chances of this happening. If I change the path to nonexisting it doesn't start the onError loop. 

I guess it could still be the case that the image itself is broken and I can think how to make sure it doesn't create a loop (i.e.  maybe do a validation before updating the src by checking if the image width/height is not 0).

Reply:
One thing you could do is check if the current src is already the src that you would otherwise set and only set it if it's different